File: LoaFavor/app/src/main/java/com/naca/loafavor/ui/fragment/EudiaFragment.java
package com.naca.loafavor.ui.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.naca.loafavor.R;
import com.naca.loafavor.data.NPC;
import com.naca.loafavor.data.NPCViewModel;
import com.naca.loafavor.databinding.FragmentEudiaBinding;
import com.naca.loafavor.data.NPCAdapter;

import java.util.ArrayList;

public class EudiaFragment extends Fragment {

    private NPCViewModel viewModel;
    private FragmentEudiaBinding binding;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        int layoutId = R.layout.fragment_eudia;
        binding = DataBindingUtil.inflate(inflater, layoutId, container, false);
        binding.setLifecycleOwner(this);

        View view = binding.getRoot();
        viewModel = new ViewModelProvider(this).get(NPCViewModel.class);
        viewModel.setList(3, 5);

        binding.eudiaRecy.setLayoutManager(new LinearLayoutManager(getActivity()));
        binding.eudiaRecy.setHasFixedSize(true);

        NPCAdapter mAdapter = new NPCAdapter(viewModel.getNPCList().getValue());
        binding.eudiaRecy.setAdapter(mAdapter);

        viewModel.getNPCList().observe(getViewLifecycleOwner(), new Observer<ArrayList<NPC>>() {
            @Override
            public void onChanged(ArrayList<NPC> npcList) {
                mAdapter.setNPCList(npcList);
            }
        });


        return view;
    }
}

File: LoaFavor/app/src/main/java/com/naca/loafavor/ui/eudia/EudiaFragment.kt
package com.naca.loafavor.ui.eudia

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.naca.loafavor.R
import com.naca.loafavor.data.NPC
import com.naca.loafavor.databinding.FragmentEudiaBinding
import com.naca.loafavor.ui.adapter.NPCAdapter
import com.naca.loafavor.ui.arthemis.EudiaViewModel
import java.util.*

class EudiaFragment : Fragment() {

    private lateinit var eudiaViewModel: EudiaViewModel

    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {
        var binding : FragmentEudiaBinding = DataBindingUtil.inflate(inflater, R.layout.fragment_eudia, container, false)
        binding.setLifecycleOwner(this)

        eudiaViewModel = ViewModelProvider(this).get(EudiaViewModel::class.java)

        var view : View = binding.root

        var NPCList = LinkedList<NPC>()
        NPCList.add(eudiaViewModel.Thunder.value!!)
        NPCList.add(eudiaViewModel.Morina.value!!)
        NPCList.add(eudiaViewModel.Urere.value!!)


        binding.eudiaRecy.layoutManager = LinearLayoutManager(activity)
        binding.eudiaRecy.setHasFixedSize(true)

        var mAdapter = NPCAdapter(NPCList)


        binding.eudiaRecy.adapter = mAdapter

        return view
    }

}